x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/AzureOpenAiSecretSettings.java

```diff
@@ -40,13 +40,6 @@ public static AzureOpenAiSecretSettings fromMap(@Nullable Map<String, Object> ma
         SecureString secureApiToken = extractOptionalSecureString(map, API_KEY, ModelSecrets.SECRET_SETTINGS, validationException);
         SecureString secureEntraId = extractOptionalSecureString(map, ENTRA_ID, ModelSecrets.SECRET_SETTINGS, validationException);
 
-        if (secureApiToken == null && secureEntraId == null) {
```

Comment:
Sorry, if my previous comment was confusing. We do need a check here right because we're extracting optional strings, so in the case where neither of them are specified we want to add a validation error, or am I missing something? What I meant is that we don't need `throw validationException;` in this `if-block` or the block below because once we add a validation exception `if (validationException.validationErrors().isEmpty() == false)` will return false and do the throw for us.

Reply:
Ah - sorry! I may have read your comment in haste... yes, we do want a validation exception here if neither are filled in... but yeah - I see exactly what you mean now.